src/gui/qgsdoublevalidator.h

```diff
@@ -0,0 +1,155 @@
+#ifndef QGSDOUBLEVALIDATOR_H
+#define QGSDOUBLEVALIDATOR_H
+
+/***************************************************************************
+                         qgsdoublevalidator.h  -  description
+                             -------------------
+    begin                : June 2020
+    copyright            : (C) 2020 by Sebastien Peillet
+    email                : sebastien.peillet@oslandia.com
+
+  adapted version of Qgslonglongvalidator + QgsFieldValidator
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <limits>
+#include <QRegExpValidator>
+#include <QLocale>
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+
+/**
+ * \ingroup gui
+ *
+ * QgsDoubleValidator is a QLineEdit Validator that combines QDoubleValidator
+ * and QRegularExpressionValidator to allow user to enter double with both
+ * local and C interpretation.
+ *
+ * \since QGIS 3.14
+ */
+class GUI_EXPORT QgsDoubleValidator : public QRegularExpressionValidator
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     * \since QGIS 3.14
+     */
+    explicit QgsDoubleValidator( QObject *parent );
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     * \since QGIS 3.14
+     */
+    QgsDoubleValidator( QRegularExpression reg, double bottom, double top, QObject *parent );
```

Comment:
```suggestion
    QgsDoubleValidator( const QRegularExpression &expression, double bottom, double top, QObject *parent );
```

Avoid abbreviations in public API

Reply:
Also add explanation of what bottom/top means

Reply:
Done !